test/readConfig.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const metatests = require('metatests');
+
+const { clone } = require('@metarhia/common');
+
+const {
+  /* eslint-disable no-unused-vars */
+  default: { decorators, ...defParam },
+  fs: { load, readConfig },
+} = require('../metaschema');
+
+const { getSchemaDir } = require('./utils');
+
+metatests.test(
+  'Must properly validate application schemas by system schemas.',
+  async test => {
+    let updatedParam, ms, modulesMs, sysMs, incorrectMs;
+    const systemParam = clone(defParam);
+    const param = clone(defParam);
+
+    const path = getSchemaDir('schemaValidation');
+    const aPath = getSchemaDir('applicationSchema');
+    const mPath = getSchemaDir('modules');
+
+    try {
+      updatedParam = await readConfig(path, systemParam, param);
+      ms = await load(aPath, updatedParam.options, updatedParam.config);
+      modulesMs = await load(mPath, updatedParam.options, updatedParam.config);
+      sysMs = await load(path, systemParam.options, systemParam.config);
+    } catch (error) {
+      test.fail(error);
+      test.end();
+      return;
+    }
+
+    test.strictSame(
+      updatedParam.config.processors.domains.hasOwnProperty('validateInstance'),
+      true
+    );
+    test.strictSame(
+      updatedParam.config.processors.category.hasOwnProperty(
+        'validateInstance'
+      ),
+      true
+    );
+
+    test.strictSame(ms.domains.size, 8);
+    test.strictSame(ms.categories.size, 5);
+
+    test.strictSame(modulesMs.domains.size, 3);
+    test.strictSame(modulesMs.categories.size, 2);
+
+    test.strictSame(
```

Comment:
I do not see a reason to test instance validation here, it is already tested in a dedicated test.